### ISSUE TITLE
Suggestion: More UGC Support

### ISSUE THREAD

**Issue by lazyRares** [Foohy/jazztronauts#133]
It would be awesome if we got more support for custom things such as new quests, or my personal interest, custom hubs! I've already started tinkering with the FGD to make it more user friendly adding nice little dropdowns and selections for entities so they aren't annoying to work with. Not to mention making my own hub right now, only problem is that certain things will cause a few errors, namely: Cutscenes, since their positioning won't be 100% because they rely on coordinates (specific ones). Finally, what I really think is killing the ability for custom hubs right now is that the trolley will always send you back to the "jazz_bar" map, but it would be nice to be able to choose which hub it goes back to by possibly storing the name of the last hub map players were on (possibly prefixed by a "jazz_" check) and sending them there instead of a hardcoded "jazz_bar".

TL:DR, Allow more modularity by removing hardcoded limits to allow support for custom hubs, and maybe quests, but that's the least of my worrys. Also, consider polishing FGD.

**Comment by SageJFox** [comment]
Added the ability to set a non-standard hub map with the convar `jazz_hub [map name]` in #166.

Not marking as closed as there's more (and very good) suggestions here too, but hopefully that helps?
